modules/RTC/RTCUtils.js

```diff
@@ -1482,11 +1502,43 @@ class RTCUtils extends Listenable {
             const umDevices = options.devices || [];
             const isDesktopDeviceRequsted = umDevices.indexOf('desktop') !== -1;
 
-            return isDesktopDeviceRequsted
-                ? this._newGetDesktopMedia(
-                    options.desktopSharingExtensionExternalInstallation,
-                    options.desktopSharingSources)
-                : Promise.resolve();
+            let getDesktopPromise;
+
+            if (isDesktopDeviceRequsted) {
```

Comment:
and what's the scenario we get here instead of the branch above?  why does the code not look quite the same as the `options.screenSharingSourceDeviceId` branch above?

Reply:
Do you mean the difference between "old gum flow" and "new gum flow"? The method here is for new gum flow. The biggest difference between the two is I didn't add the "constraints.video.deviceId.exact" constraints above in the old gum flow because I wasn't confident of my implementation here and if I should just copy/paste (or move out) the logic so old gum flow also uses constraints.video.deviceId.exact; the biggest issue there being the old getConstraints has to be modified to handle having deviceId.exact.